### PR TITLE
pkg/k8s: properly handle empty NamespaceSelector

### PR DESCRIPTION
**Pull request by ianvernon**
If a rule is passed with an empty NamespaceSelector, this means that all traffic should be
allowed to/from all pods in all namespaces. However, Cilium was improperly treating this as
allowing all traffic to/from all destinations/sources accordingly.

Signed-off by: Ian Vernon <ian@cilium.io>

```release-note
properly handle empty NamespaceSelector in Kubernetes NetworkPolicy
```

Fixes: #4513

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5250)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Build failure: 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/5742

* Issues were due to accessing YAML files from GitHub (not related to changes, are infrastructural related)
* Another issue was due to `RunInit` failure found in logs - this is during the kvstore test (not related to my changes))

Trying again.

**Comment by ianvernon** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Comment by ianvernon** [comment]
Prior test failures were due to #5251 . Rebased, and trying again.

**Comment by ianvernon** [comment]
test-me-please

File: pkg/k8s/network_policy.go [hunk 1]
```diff
@@ -24,8 +24,15 @@ import (
 	"github.com/cilium/cilium/pkg/logging/logfields"
 	"github.com/cilium/cilium/pkg/policy"
 	"github.com/cilium/cilium/pkg/policy/api"
-
 	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	allowAllNamespacesRequirement = v1.LabelSelectorRequirement{
```
Comment (raybejjani) [inline pkg/k8s/network_policy.go:32]: I don't know how to emoji a single line but 💯

**Comment by ianvernon** [comment]
K8s upstream tests failed when downloading the VM: https://jenkins.cilium.io/job/Cilium-PR-Kubernetes-Upstream/492/execution/node/32/log/

Failure is unrelated to my changes.

**Comment by ianvernon** [comment]
test-upstream-k8s

**Pull request merged**